Exercise 2. Sentiment Analysis

What is the overall sentiment of the ‘spaceX.txt’ text? Provide the compound, negative,
neutral, and positive metrics. Is the text more ‘subjective’ or more ‘objective’?

In [4]:
from textblob import TextBlob


In [6]:
pip install vaderSentiment


     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 126.0/126.0 kB 3.7 MB/s eta 0:00:00


In [7]:
from vaderSentiment.vaderSentiment import SentimentIntensityAnalyzer


In [8]:

#  'spaceX.txt' file
text = """
SpaceX CEO Elon Musk said on Friday that he would activate the firm's satellite internet service, Starlink, in response
to U.S. Secretary of State Antony Blinken's tweet that the United States took action "to advance internet freedom and the
free flow of information" to Iranians. The U.S. Treasury Department on Friday issued guidance expanding internet services
available to Iranians despite U.S. sanctions on the country, amid protests around Iran following the death of a 22-year-old
woman in custody. A Treasury official briefing reporters said: "Our understanding of Starlink is that what they provide
would be commercial grade, and it would be hardware that's not covered in the general license; so that would be something
they would need to write into Treasury for." A U.S. State Department spokesperson later said of Friday's updated license
that it was self-executing and that "anyone who meets the criteria outlined in this general license can proceed with their
activities without requesting additional permissions." Musk could not be reached for comments or clarification regarding
Starlink's clearance to operate in Iran. Iranians have been protesting over the death of Mahsa Amini last week while in
police custody after being arrested by the morality police for wearing "unsuitable attire". Musk said on Monday that the
company wanted to provide Starlink satellite broadband service - already provided to Ukraine for its fight against Russia's
invasion - to Iranians, and would ask for a sanctions exception. The U.S. State Department spokesperson added that if SpaceX
were to determine that some activity aimed at Iranians requires a specific license, "OFAC would welcome it and prioritize
it". "By the same token, if SpaceX determines that its activity is already authorized and has any questions, OFAC also
welcomes that engagement," the State Department spokesperson said.
"""



In [9]:
vader_analyzer = SentimentIntensityAnalyzer()
vader_sentiment = vader_analyzer.polarity_scores(text)

text_blob = TextBlob(text)
subjectivity = text_blob.sentiment.subjectivity

vader_sentiment, subjectivity

({'neg': 0.061, 'neu': 0.863, 'pos': 0.076, 'compound': 0.5999},
 0.30897435897435904)

A subjectivity score of approximately 0.309, which is close to 0, indicates that the text is primarily objective. This means it's based on facts, data, and events, rather than personal feelings or opinions.